splink/linker.py

```diff
@@ -523,13 +526,16 @@ def _instantiate_comparison_levels(self, settings_dict):
         instances are instead replaced with ComparisonLevels
         """
         dialect = self._sql_dialect
-        # TODO: handle case where dict is in fact a Comparison object
-        for comparison_dict in settings_dict["comparisons"]:
-            comparison_levels = comparison_dict["comparison_levels"]
-            for idx, level in enumerate(comparison_levels):
-                # if we have a ComparisonLevelCreator
-                if not isinstance(level, dict):
-                    comparison_levels[idx] = level.get_comparison_level(dialect)
+        comparisons = settings_dict["comparisons"]
```

Comment:
This key is not guaranteed to exist
```suggestion
        if "comparisons" not in settings_dict:
            return
        comparisons = settings_dict["comparisons"]
```

Reply:
Ah yep, good shout. Yeah this bit of the code will definitely need revising / some more careful thought - just had something simple that mostly works, as we may end up changing how some of this linker code works more generally